app/src/main/java/org/mozilla/focus/fragment/DownloadsFragment.java

```diff
@@ -39,6 +39,22 @@ public static DownloadsFragment newInstance() {
     public View onCreateView(LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
         recyclerView = (RecyclerView) inflater.inflate(R.layout.fragment_downloads, container, false);
+
+        recyclerView.addOnScrollListener(new RecyclerView.OnScrollListener() {
+            @Override
+            public void onScrolled(RecyclerView recyclerView, int dx, int dy) {
+                super.onScrolled(recyclerView, dx, dy);
```

Comment:
The onScrolled is fired infinitely, we shall solve this (which should be on my side) first before we move on.